src/timeutils.cc

```diff
@@ -0,0 +1,184 @@
+#include "winutils.h"
+#include <winsock2.h> //for timeval
+#include<time.h>
+
+#include <iomanip>
+#include <sstream>
+#include <sys/time.h>
+#include <cassert>
+
+#define BUF_SIZE 128
+//C++ 11: Lot of OS specific functions can be replaced by using the boost::chrono library
+
+bool ValidClockId(clockid_t clk_id) {
+    if (clk_id == CLOCK_MONOTONIC || clk_id == CLOCK_REALTIME)
+        return true;
+    else
+        return false;
+}
+
+int clock_getres_monotonic( struct timespec *res) {
+    if (res != nullptr) {
+        LARGE_INTEGER frequency;
+        if (QueryPerformanceFrequency(&frequency) != 0) {
+            double nanosecdbl = 1000000000.0 / frequency.QuadPart;
+            res->tv_nsec = static_cast<long>(nanosecdbl) % 1000000000;
```

Comment:
Please also note that `static_cast<long>(nanosecdbl)`  truncates the value, so we don't really benefit from the double precision floating point value. We may either round the result (`static_cast<uint64_t>(nanosecdbl + 0.5)`) or get rid of doubles (`uint64_t ticks_per_nanosec = 1000000000 / frequency.QuadPart;`).